presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/EvaluateZeroLimit.java

```diff
@@ -27,10 +28,18 @@
 public class EvaluateZeroLimit
     implements Rule
 {
+    private static final Pattern PATTERN = Pattern.node(LimitNode.class);
+
+    @Override
+    public Pattern getPattern()
+    {
+        return PATTERN;
+    }
+
     @Override
     public Optional<PlanNode> apply(PlanNode node, Lookup lookup, PlanNodeIdAllocator idAllocator, SymbolAllocator symbolAllocator)
     {
-        if (!(node instanceof LimitNode)) {
+        if (!PATTERN.matches(node)) {
```

Comment:
I would love to do it as you suggest, but I think it is a design decision to be make here.

Do we impose the contract that in order to call `apply` you need to check if pattern matches first?

I could hide this boiler plate under some abstract rule, but I would prefer to know the answer for the above question first.